### PR TITLE
CLOUDP-128547: Add cloud provider access role feature

### PR DESCRIPTION
**Pull request by Sugar-pack**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

**Review by igor-karpukhin — APPROVED** [review]
Looks good 👍 @fabritsius should we merge your feature first prior to merging this PR?

File: pkg/controller/atlasproject/cloud_provider_access.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+		return result
+	}
+	context.SetConditionTrue(status.CloudProviderAccessReadyType)
+	return result
+}
+
+func syncProviderAccessStatus(customContext *workflow.Context, specs []v1.CloudProviderAccessRole, statuses []status.CloudProviderAccessRole, groupID string) (workflow.Result, status.ConditionType) {
+	client := customContext.Client
+	logger := customContext.Log
+	ctx := context.Background() // TODO: eoaueoaueoa
```
Comment (fabritsius) [inline pkg/controller/atlasproject/cloud_provider_access.go:31]: This line seems unfinished at the end part 😂

Comment (Sugar-pack) [reply pkg/controller/atlasproject/cloud_provider_access.go]: Fixed

File: pkg/controller/atlasproject/cloud_provider_access.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+func ensureProviderAccessStatus(ctx context.Context, customContext *workflow.Context, project *v1.AtlasProject, groupID string) workflow.Result {
+	roleStatuses := project.Status.DeepCopy().CloudProviderAccessRoles
+	roleSpecs := project.Spec.DeepCopy().CloudProviderAccessRoles
+
+	result, condition := syncProviderAccessStatus(ctx, customContext, roleSpecs, roleStatuses, groupID)
+	if result != workflow.OK() {
+		customContext.SetConditionFromResult(condition, result)
+		return result
+	}
+	customContext.SetConditionTrue(status.CloudProviderAccessReadyType)
```
Comment (fabritsius) [inline pkg/controller/atlasproject/cloud_provider_access.go:24]: `UnsetCondition` should probably be somewhere close by :)

**Review by fabritsius — APPROVED** [review]
Looks Great 👍 
Made a few tiny suggestions

**Comment by fabritsius** [comment]
@igor-karpukhin I think it can be merged as is, no need to wait :)

File: pkg/controller/atlasproject/cloud_provider_access.go (hunk 2 above)
Comment (Sugar-pack) [reply pkg/controller/atlasproject/cloud_provider_access.go:29]: Done. Thank you)

**Pull request merged**